addons/binding/org.openhab.binding.honeywellwifithermostat/src/main/java/org/openhab/binding/honeywellwifithermostat/handler/honeywellWifiThermostatHandler.java

```diff
@@ -0,0 +1,172 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.honeywellwifithermostat.internal.data.HoneywellThermostatData;
+import org.openhab.binding.honeywellwifithermostat.internal.data.HoneywellThermostatFanMode;
+import org.openhab.binding.honeywellwifithermostat.internal.data.HoneywellThermostatSystemMode;
+import org.openhab.binding.honeywellwifithermostat.internal.webapi.HoneywellWebsiteJetty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link honeywellWifiThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author JD Steffen - Initial contribution
+ */
+public class honeywellWifiThermostatHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(honeywellWifiThermostatHandler.class);
+
+    HoneywellWebsiteJetty webapi;
+    ScheduledFuture<?> refreshJob;
+
+    private String deviceID = null;
+
+    private HoneywellThermostatData thermodata;
+
+    public honeywellWifiThermostatHandler(Thing thing) {
+        super(thing);
+
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case COOL_SETPOINT:
+                DecimalType coolSP_val = (DecimalType) command;
+                thermodata.setCoolSetPoint(coolSP_val.intValue());
+                break;
+            case HEAT_SETPOINT:
+                DecimalType heatSP_val = (DecimalType) command;
+                thermodata.setHeatSetPoint(heatSP_val.intValue());
+                break;
+            case SYSTEM_MODE:
+                StringType sysmode_val = (StringType) command;
+                thermodata.setCurrentSystemMode(HoneywellThermostatSystemMode.valueOf(sysmode_val.toString()));
+                break;
+            case FAN_MODE:
+                OnOffType fanmode_val = (OnOffType) command;
+                if (command == OnOffType.OFF) {
+                    thermodata.setCurrentFanMode(HoneywellThermostatFanMode.AUTO);
+                } else if (command == OnOffType.ON) {
+                    thermodata.setCurrentFanMode(HoneywellThermostatFanMode.ON);
+                }
+
+                break;
+        }
+
+        if (!webapi.submitThermostatChange(deviceID, thermodata)) {
+            logger.error("Failed to submit changes to honeywell site.");
```

Comment:
remove the error log - we already have logging for the thing status update, which carries the same information.